app/scripts/directives/loading.dir.js

```diff
@@ -0,0 +1,27 @@
+'use strict';
+
+angular.module('yourStlCourts').directive('loading', function($timeout){
```

Comment:
Rather than focus on state changes, we should use an $httpInterceptor with a request counter so we handle network calls only. I have tried and true code from a couple other projects to use for this. Lets ditch this PR even though it was a good effort.